### PR TITLE
Add stackdriver endpoint flag

### PR DESCRIPTION
**Pull request by znirzej**

### CODE REVIEW

**Comment by znirzej** [comment]
@serathius @cezarygerard

File: custom-metrics-stackdriver-adapter/adapter.go [hunk 1]
```diff
@@ -82,6 +85,14 @@ func (sa *StackdriverAdapter) makeProviderOrDie(o *stackdriverAdapterServerOptio
 	if err != nil {
 		klog.Fatalf("Failed to create Stackdriver client: %v", err)
 	}
+	if o.StackdriverEndpoint != "" {
+		_, err = url.ParseRequestURI(stackdriverService.BasePath)
```
Comment (serathius) [inline custom-metrics-stackdriver-adapter/adapter.go:89]: o.StackdriverEndpoint?

File: custom-metrics-stackdriver-adapter/adapter.go [hunk 2]
```diff
@@ -82,6 +85,14 @@ func (sa *StackdriverAdapter) makeProviderOrDie(o *stackdriverAdapterServerOptio
 	if err != nil {
 		klog.Fatalf("Failed to create Stackdriver client: %v", err)
 	}
+	if o.StackdriverEndpoint != "" {
```
Comment (serathius) [inline custom-metrics-stackdriver-adapter/adapter.go:88]: Can you add a simple test for this validation?

File: custom-metrics-stackdriver-adapter/adapter_test.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+func TestURLValidator(t *testing.T) {
+	for url, correct := range urltests {
+		t.Run(url, func(t *testing.T) {
+			result := validateUrl(url)
+			if result != correct {
+				t.Errorf("for url %v got %v, want %v", url, result, correct)
+			}
+		})
+	}
+}
```
Comment (serathius) [inline custom-metrics-stackdriver-adapter/adapter_test.go:47]:
```suggestion
func TestURLValidator(t *testing.T) {
    for _, tc := []struct{
      url string
      expect bool
    } {
     {
       url: "https://monitoring.googleapis.com/",
       expect: true
     }
    }
		t.Run(tc.url, func(t *testing.T) {
			result := validateUrl(tc.url)
			if result != tc.expect {
				t.Errorf("for url %v got %v, expect %v", tc.url, result, expect)
			}
		})
	}
}
```
Let's use table tests, 

FYI No need to pass `expect: false` as bool default is false.

**Comment by serathius** [comment]
LGTM 
/cc @kawych

**Pull request merged**